src/commands/mod.rs

```diff
@@ -1,4 +1,34 @@
+use std::fmt::{Display, Formatter};
+use std::io;
+use std::path::PathBuf;
+
 pub(crate) mod generate_buildpack_matrix;
 pub(crate) mod generate_changelog;
 pub(crate) mod prepare_release;
 pub(crate) mod update_builder;
+
+pub(crate) fn resolve_path(value: Option<PathBuf>) -> Result<PathBuf, ResolvePathError> {
+    std::env::current_dir()
+        .map_err(ResolvePathError)
+        .map(|current_dir| {
+            if let Some(path) = value {
+                if path.is_absolute() {
+                    path
+                } else {
+                    current_dir.join(path)
+                }
+            } else {
+                current_dir
+            }
+        })
```

Comment:
Just as a side-note: I think this implementation would be more readable in this form:

```rust
match value {
    None => std::env::current_dir(),
    Some(path) if path.is_absolute() => Ok(path),
    Some(path) => std::env::current_dir().map(|current_dir| current_dir.join(path))
}.map_err(ResolvePathError)
```

Ideal functions (pure functions) are just mappings from the input values to the output values. In simple cases like this where we can actually spell out the mappings explicitly, `match` is a good tool for doing so, even when the function is not pure like this one. I'm mentioning the mapping analogy as it has been a useful mental shortcut for me when thinking about code. :)